sequence_processing_pipeline/Pipeline.py

```diff
@@ -459,21 +459,50 @@ def get_sample_names(self, project_name=None):
 
     def _get_sample_names_from_sample_sheet(self, project_name):
         if project_name is None:
-            return [x.Sample_Name for x in self.sample_sheet.samples]
+            if 'orig_name' in self.sample_sheet.samples[0]:
+                # return set as a list to match non-replicate results.
+                return sorted(list({x.orig_name for x in
+                                    self.sample_sheet.samples}))
+            else:
+                return [x.Sample_Name for x in self.sample_sheet.samples]
```

Comment:
Why are these not sorted?

Reply:
In a sheet w/out replicates, each sample-name should be unique and the results are returned in order of appearance in the sample-sheet. With replicates, the order will be unpredictable after we convert the list into a set; this makes it harder to compare observed results against expected ones hence the sorted order.

Reply:
Right but the expectations to the caller now differ based on arguments

Reply:
I can't argue with that, especially if I'm casting it into a list to preserve expectations.